pkg/bwagreement/server.go

```diff
@@ -106,3 +112,67 @@ func (s *Server) BandwidthAgreements(ctx context.Context, rba *pb.RenterBandwidt
 	s.logger.Debug("Stored Agreement...")
 	return reply, nil
 }
+
+func (s *Server) verifySignature(ctx context.Context, rba *pb.RenterBandwidthAllocation) error {
```

Comment:
maybe move verifySignature() into auth, and just give it a different name than the existing function there?

Reply:
I will have to think about this as this function, needs context, satellite DB arguments, bwagreement.Server. Am afraid there would be bit of repetition. For now, I recommend to keep it bit separate and when we unify the signature verification, we can consolidate these two functions.